### ISSUE TITLE
transaction monitor is not catching modifying CTEs (Postgres)

### ISSUE THREAD

**Issue by rumbin** [dbeaver/dbeaver#7445]
#### System information: 
- Operating system: Win 10
- DBeaver version: 6.3.0.201912042122
- Additional extensions: none

#### Connection specification:
- Database name and version: PostgreSQL 10, 11, 12
- Driver name: postgresql-42.2.5.jar
- Do you use tunnels or proxies (SSH, SOCKS, etc): no

#### Describe the problem you're observing:
Background:  
* The transaction monitor is supposed to count modifying statements, turning green in case of uncommitted modifing statements.
* Postgres supports so-called _modifying CTEs_ where a modifying statement (INSERT, UPDATE, DELETE) can be embedded into a CTE. Even multiple modifying CTEs can exist in a single statement.

Issue:  
* The transaction monitor does not catch and display any modifying CTEs, when the main statement is a SELECT.
* Thus, there may exist uncommitted modifying statements that are not visible to the user.

#### Steps to reproduce, if exist:
* Switch to manual commit mode
* Run any statement that includes a modifying CTE
* Check the transaction monitor: it will not display any modified statement
* Rollback maybe :-)

Example:

```
with del as (
	delete
	from
		mytable
	returning *
)
select
	*
from
	del
```

**Comment by serge-rider** [comment]
Yes, we need better detection of "transaction affecting queries". Probably checking query results (e.g. affected rows) is the best way to do it.

**Comment by kseniiaguzeeva** [comment]
Should CTE transactions (with the main statement "SELECT") be added to modifying statements counter?
![blg7v5PI2h](https://user-images.githubusercontent.com/51405061/71559541-635d8b00-2a70-11ea-85ac-055a258d62b9.gif)

**Comment by rumbin** [comment]
In my eyes _modifying_ CTEs should definitely be added to the modifying statements counter. Otherwise the users might assume that they could safely commit or rollback without side effects after running such a statement.
This is dangerous.

**Comment by serge-rider** [comment]
Now (in addition to query examine) we use JDBC API to determine number of affected rows.
Unfortunately PG JDBC driver doesn't return affected row count in case of CTE queries which embed DML statements.
Dunno how can we handle this..
